src/stirling/source_connectors/socket_tracer/conn_tracker.cc

```diff
@@ -1024,13 +1046,21 @@ void ConnTracker::InferConnInfo(system::ProcParser* proc_parser,
 
   // Success! Now copy the inferred socket information into the ConnTracker.
 
-  Status s = ParseSocketInfoAddr(socket_info, &open_info_.remote_addr, &open_info_.local_addr);
+  Status s = ParseSocketInfoLocalAddr(socket_info, &open_info_.local_addr);
   if (!s.ok()) {
-    conn_resolver_.reset();
-    conn_resolution_failed_ = true;
-    LOG(ERROR) << absl::Substitute("Remote and local address (type=$0) parsing failed. Message: $1",
-                                   socket_info.family, s.msg());
-    return;
+    CONN_TRACE(2) << absl::Substitute("Local address (type=$0) parsing failed. Message: $1",
+                                      socket_info.family, s.msg());
+  }
+  if (!parse_local_addr_only) {
+    Status s = ParseSocketInfoRemoteAddr(socket_info, &open_info_.remote_addr);
+    if (!s.ok()) {
+      // reset only if remote address is not found
+      conn_resolver_.reset();
+      conn_resolution_failed_ = true;
+      CONN_TRACE(2) << absl::Substitute("Remote address (type=$0) parsing failed. Message: $1",
+                                        socket_info.family, s.msg());
+      return;
+    }
```

Comment:
For a bit of additional symmetry, would it make sense to write this as:

```
if (open_info_.local_addr.empty()) { // empty() is pseudo-code
  Status s = ParseSocketInfoLocalAddr(socket_info, &open_info_.local_addr);
  ...
}

if (open_info_.remote_addr.empty()) { // empty() is pseudo-code
  Status s = ParseSocketInfoRemoteAddr(socket_info, &open_info_.local_addr);
  ...
}
```

Also, wondering again about why local and remote addresses are treated differently in terms of `conn_resolution_failed_` treatment. Have you seen cases where remote addresses can be inferred by not local ones?

Reply:
Thanks for the suggestion @oazizi000. I've just updated the conditions for simplicity. The main motivation behind treating local/remote addresses differently is to prevent the connection from being dropped when we already have a remote endpoint, but fail to parse a local address. See previous comment for additional details.